spec/requests/file_attachments_spec.rb

```diff
@@ -298,18 +298,27 @@
       expect(response).to redirect_to(featured_attachments_path(edition.document))
     end
 
-    it "returns issues and an unprocessable response when there are requirement issues" do
+    it "returns unprocessable response and highlights errors when requirements not met" do
       max_length = Requirements::FileAttachmentMetadataChecker::UNIQUE_REF_MAX_LENGTH
       too_long_unique_reference = "a" * (max_length + 1)
 
       patch edit_file_attachment_path(edition.document,
                                       file_attachment_revision.file_attachment_id),
-            params: { file_attachment: { unique_reference: too_long_unique_reference } }
+            params: { file_attachment: { isbn: "foo", unique_reference: too_long_unique_reference } }
 
       expect(response).to have_http_status(:unprocessable_entity)
+
+      expect(response.body).to have_content(
+        I18n.t!("requirements.file_attachment_isbn.invalid.form_message"),
+      )
+      expect(response.body)
+        .to have_selector("input[name='file_attachment[isbn]'].govuk-input--error")
```

Comment:
Ooh, this is interesting: we don't actually test for highlighted errors anywhere else in the app. I notice that we also don't check that user input is preserved after such an error.

I was wondering if these checks should be in the request test, and it seems reasonable, because otherwise we're not covering [this line](https://github.com/alphagov/content-publisher/blob/master/app/controllers/file_attachments_controller.rb#L134). However, they could get quite verbose.

Since these checks are inconsistent with all the other tests for unprocessable responses, would you be willing to write a pain card about these (two) issues, in favour of these checks?

Reply:
Yes, it was difficult to know where to put these tests as there wasn't really a precedent. Are you suggesting keeping the tests I've added here (and indeed adding to them, to test that the incorrect values are retained), then writing a pain card so that similar tests can be applied everywhere else? Or are you suggesting removing the tests from this PR?

Reply:
I think we should remove the tests for now, as opposed to putting more effort into making them complete in this one place. Since none of these behaviours - the additional highlighting, the preservation of values - is critical for users, I'm not too worried if such things continue to slip through for the time being, provided we have a card to address the issue in general. Would you like to write a pain card, if you agree?

Reply:
Here is the pain card: https://trello.com/c/hccZ9cuf/1715-we-dont-test-highlighted-error

I've now removed the tests 👍